File: HomeServer/dao/roomdao.py
from database.databaseconfig import *
from model.room import Room


class RoomDao(object):

    @db_session
    def create_room(self, name):
        return Room(name=name)

    @db_session
    def read_rooms(self):
        return Room.select()[:]

    @db_session
    def read_room(self, room_id):
        return Room[room_id]

    @db_session
    def update_room(self, room_id, name):
        r = Room[room_id]
        r.name = name
        return r

    @db_session
    def delete_room(self, room_id):
        Room[room_id].delete()

    @db_session
    def get_room_module_configurations(self, room_id):
        return list(Room[room_id].module_configurations)

File: MobileApplication/app/src/main/java/pl/polsl/mateusz/chudy/mobileapplication/dao/RoomDao.kt
package pl.polsl.mateusz.chudy.mobileapplication.dao

import android.arch.persistence.room.*
import pl.polsl.mateusz.chudy.mobileapplication.model.Room

/**
 *
 */
@Dao
interface RoomDao {

    @Query("select * from room")
    fun getRooms(): List<Room>

//    @Query("select * from room where room_id = :roomId")
//    fun getRoom(roomId: Long): Room
//
//    @Insert
//    fun createRoom(room: Room)
//
//    @Update
//    fun updateRoom(room: Room)

//    @Query("delete from room where room_id = :roomId")
//    fun deleteRoom(roomId: Long)

    @Delete
    fun deleteRoom(room: Room)
}